### PR TITLE
make mmap tests friendlier to containers

### PR DESCRIPTION
**Pull request by wfurt**
Even if it is possible to construct container in a way this test pass, that is not the case even when started with `--privileged` flag.

This change makes it easier to run MemoryMappedFiles tests inside containers. 
If privilege is not available it will report test skip instead of failure. 

```
helixbot@eea8b9b5c8fa:/mnt/wfurt-corefx-arm/artifacts/bin/tests/System.IO.MemoryMappedFiles.Tests/netcoreapp-Linux-Debug-arm$ ../../../testhost/netcoreapp-Linux-Debug-arm/dotnet xunit.console.dll System.IO.MemoryMappedFiles.Tests.dll -xml testResults.xml -nologo -notrait category=nonnetcoreapptests -notrait category=nonlinuxtests -notrait category=failing
 Discovering: System.IO.MemoryMappedFiles.Tests (method display = ClassAndMethod, method display options = None)
  Discovered:  System.IO.MemoryMappedFiles.Tests (found 80 of 103 test cases)
  Starting:    System.IO.MemoryMappedFiles.Tests (parallel test collections = on, max threads = 6)
    System.IO.MemoryMappedFiles.Tests.MemoryMappedViewStreamTests.ValidAccessLevelCombinations(mapAccess: ReadWriteExecute, viewAccess: ReadWriteExecute) [SKIP]
      Execute permission failing in container.
    System.IO.MemoryMappedFiles.Tests.MemoryMappedViewStreamTests.ValidAccessLevelCombinations(mapAccess: ReadWriteExecute, viewAccess: ReadExecute) [SKIP]
      Execute permission failing in container.
    System.IO.MemoryMappedFiles.Tests.MemoryMappedViewAccessorTests.ValidAccessLevelCombinations(mapAccess: ReadWriteExecute, viewAccess: ReadWriteExecute) [SKIP]
      Execute permission failing in container.
    System.IO.MemoryMappedFiles.Tests.MemoryMappedViewAccessorTests.ValidAccessLevelCombinations(mapAccess: ReadWriteExecute, viewAccess: ReadExecute) [SKIP]
      Execute permission failing in container.
  Finished:    System.IO.MemoryMappedFiles.Tests
=== TEST EXECUTION SUMMARY ===
   System.IO.MemoryMappedFiles.Tests  Total: 345, Errors: 0, Failed: 0, Skipped: 4, Time: 5.316s
```

### CODE REVIEW

File: src/System.IO.MemoryMappedFiles/tests/MemoryMappedViewAccessor.Tests.cs [hunk 1]
```diff
@@ -56,7 +57,7 @@ public void InvalidArguments()
             }
         }
 
-        [Theory]
+        [ConditionalTheory]
```
Comment (danmosemsft) [inline src/System.IO.MemoryMappedFiles/tests/MemoryMappedViewAccessor.Tests.cs]: What does `ConditionalTheory` do? Is it necessary to have, if you might throw `SkipTestException`?

**Comment by danmosemsft** [comment]
Do you know why containers are special in this way? Mostly they try to allow code to run unmodified, I assume.

File: src/System.IO.MemoryMappedFiles/tests/MemoryMappedViewAccessor.Tests.cs (hunk 1 above)
Comment (wfurt) [reply src/System.IO.MemoryMappedFiles/tests/MemoryMappedViewAccessor.Tests.cs]: yes. ConditionalTheory is extension handling SkipTestException

**Comment by wfurt** [comment]
yes, it was discussed in #34718. It is related to where and how /dev is mounted.

File: src/System.IO.MemoryMappedFiles/tests/MemoryMappedViewAccessor.Tests.cs (hunk 1 above)
Comment (safern) [reply src/System.IO.MemoryMappedFiles/tests/MemoryMappedViewAccessor.Tests.cs]: I didn't even know you could use ConditionalTheory without parms and make it functional :smile:

Comment (wfurt) [reply src/System.IO.MemoryMappedFiles/tests/MemoryMappedViewAccessor.Tests.cs]: that was port of change I did few month back to support dynamic skip e.g. skip after test method is executed. That is useful for cases when you cannot determine it as static property - like curl vs Socket HTTP handler. In this case one could detect the ability and we could split the test to two. but I think this is simpler.

File: src/System.IO.MemoryMappedFiles/tests/MemoryMappedViewAccessor.Tests.cs [hunk 2]
```diff
@@ -80,10 +81,21 @@ public void ValidAccessLevelCombinations(MemoryMappedFileAccess mapAccess, Memor
             AssertExtensions.ThrowsIf<IOException>(PlatformDetection.IsUap && mapAccess == MemoryMappedFileAccess.ReadWriteExecute && viewAccess == MemoryMappedFileAccess.ReadWriteExecute,
             () =>
             {
-                using (MemoryMappedFile mmf = MemoryMappedFile.CreateNew(null, Capacity, mapAccess))
-                using (MemoryMappedViewAccessor acc = mmf.CreateViewAccessor(0, Capacity, viewAccess))
+                try {
```
Comment (stephentoub) [inline src/System.IO.MemoryMappedFiles/tests/MemoryMappedViewAccessor.Tests.cs]: Nit: please move the { to the next line

**Pull request merged**